src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenceResourceTest.java

```diff
@@ -243,6 +245,47 @@ public void testUpdateOffence() {
         }
     }
 
+    @Nested
+    @DisplayName("Tests linking and unlinking of offences to schedules")
+    public class LinkOffencesToSchedulesTests {
+        private final String maintainerTokenSchedule = authTokenHelper.getToken(AuthToken.UPDATE_OFFENCE_SCHEDULES);
+
+        @Sql(scripts = {"/sql/create_offence_data.sql"},
+            executionPhase = ExecutionPhase.BEFORE_TEST_METHOD,
+            config = @SqlConfig(transactionMode = TransactionMode.ISOLATED))
+        @Sql(scripts = {"/sql/clean_offences.sql"},
+            executionPhase = ExecutionPhase.AFTER_TEST_METHOD,
+            config = @SqlConfig(transactionMode = TransactionMode.ISOLATED))
+        @Test
+        public void testLinkAndUnlinkingOffences() {
+            String offenceToScheduleMappings = "[" +
+                "{ \"offenceCode\": \"COML025\", \"schedule\": \"SCHEDULE_15\" }," +
+                "{ \"offenceCode\": \"STAT001\", \"schedule\": \"SCHEDULE_13\" }," +
+                "{ \"offenceCode\": \"RC86355\", \"schedule\": \"SCHEDULE_13\" }" +
+                "]";
+            final var offencesToSchedules = createHttpEntity(maintainerTokenSchedule, offenceToScheduleMappings);
+            final var response = postRequest(offencesToSchedules, "/api/offences/link-to-schedule");
+            assertThatStatus(response, 201);
+
+            final var requestEntity = createHttpEntityWithBearerAuthorisation("RO_USER", List.of("ROLE_VIEW_PRISONER_DATA"), Map.of());
+            final var bookingResponse = getBookingResponse(requestEntity);
+            assertThatJsonFileAndStatus(bookingResponse, HttpStatus.OK.value(), "sentences-and-offences-details-with-schedule.json");
+
+            final var unlinkingResponse = postRequest(offencesToSchedules, "/api/offences/unlink-from-schedule");
+            assertThatStatus(unlinkingResponse, 200);
+            final var bookingResponseAfterUnlinking = getBookingResponse(requestEntity);
+            assertThatJsonFileAndStatus(bookingResponseAfterUnlinking, HttpStatus.OK.value(), "sentences-and-offences-details.json");
+        }
+    }
+
+    private ResponseEntity<String> getBookingResponse(HttpEntity<?> requestEntity) {
+        return testRestTemplate.exchange("/api/offender-sentences/booking/-20/sentences-and-offences",
```

Comment:
you could use webTestClient (that is preferred for new tests)

Reply:
agreed and done